sdk/component/conf.go

```diff
@@ -110,9 +100,9 @@ func (rc *RunnerConfig) isValid() error {
 			FieldName: "panic_handler",
 			Reason:    errReasonCannotBeNil,
 		}
-	case rc.storerConfig.store == nil:
+	case rc.Storer == nil:
```

Comment:
this check is not enough because an interface could be pointing to a Nil value but itself is not Nil. the correct check is something like this:
```golang
return rc.Storer == nil || (reflect.ValueOf(rc.Storer).Kind() == reflect.Ptr && reflect.ValueOf(rc.Storer).IsNil())
```
we can move this into an independent function so that it's used in other places too

Reply:
Done